lib/slanger/channel.rb

```diff
@@ -24,15 +24,17 @@ def from(channel_id)
       end
 
       def lookup(channel_id)
-        all.detect { |o| o.channel_id == channel_id }
+        all[channel_id]
```

Comment:
Lookup was only try to find the first one and increase or decrease the connection count as needed.
So we do not need to use array, we could use hash directly and only keep one channel per channel id.

Reply:
The list of message consumers is maintained by PubsubClient not here. Here is just a channel list.